### ISSUE TITLE
Set up CI on GitHub Actions for Windows virtual environment

### ISSUE THREAD

**Issue by bmuschko** [bmuschko/gradle-docker-plugin#938]
Ensure that the plugin is compatible in Windows environments. Initial work has been done in [this GitHub Actions workflow](https://github.com/bmuschko/gradle-docker-plugin/blob/master/.github/workflows/windows-build.yml). Fix any tests to may fail on Windows.

**Comment by nirud** [comment]
Hi,

as you requested i can post the failure logs. 
The first failure is about escaping the path in the tests, i attachted to log file:

[log.txt](https://github.com/bmuschko/gradle-docker-plugin/files/4694454/log.txt)

**Comment by bmuschko** [comment]
If someone from the community wants to make more progress on the Windows-based GitHub Actions workflow: https://github.com/bmuschko/gradle-docker-plugin/blob/master/.github/workflows/disabled/windows-build.yml. So far I couldn't get it to work.

**Comment by bmuschko** [comment]
@watzerm This might be a good first issue to pick up. It will benefit us a lot down the road to ensure that the plugin functionality works properly on Linux, MacOS and Windows. Any thoughts?

**Comment by astudentofblake** [comment]
To test my patch I got the tests running locally in windows, with docker desktop running linux containers.
 - line ending fixes
 - must pass through environment variables (causes the resolution failures)
 - some tests don't apply
so I tried running in the GitHub Actions, using the windows-build.yaml

The GitHub Hosted runner windows-2019 runs docker hosting windows containers. It cannot be switched to linux containers (as I understand this would need hyper-v). I don't think GitHub Actions lets me run both a windows runner and a linux runner at the same time, and connect between them.
Most of the tests, and part of the functionality rely on linux containers (alpine).

The functional tests cannot be run then on the windows-2019 runner. I think it may be worth adding the windows build with just the build and unit test steps, to allow local testing more easily on windows. Possibly a subset of tests may be added later, using a different base container if you want to support this. 
@bmuschko 
Would it be worth delivering a PR for the windows fixes and the build + unit test steps?
